framework/validators/IpValidator.php

```diff
@@ -0,0 +1,555 @@
+
+    /**
+     * @const integer is used to change check order by [[isAllowed]]
+     * @see isAllowed()
+     * @see order
+     */
+    const ORDER_ALLOW_DENY = 0;
+
+    /**
+     * @const integer is used to change check order by [[isAllowed]]
+     * @see isAllowed()
+     * @see order
+     */
+    const ORDER_DENY_ALLOW = 1;
+
+    /**
+     * @const string default negation char when [[negationChar]] is set to `true`
+     * @see negationChar
+     */
+    const DEFAULT_NEGATION_CHAR = '!';
+
+    /**
+     * @var boolean whether support of IPv6 addresses is enabled
+     */
+    public $ipv6 = true;
+
+    /**
+     * @var boolean whether support of IPv4 addresses is enabled
+     */
+    public $ipv4 = true;
+
+    /**
+     * @var boolean whether the address can be a CIDR subnet
+     *    true - the subnet is required
+     *   false - the address can not have the subnet
+     *    null - ths subnet is optional
+     */
+    public $subnet = false;
+
+    /**
+     * @var boolean whether to add the prefix with the smallest length (32 for IPv4 and 128 for IPv6)
+     * to an address without it.
+     * Works only when attribute [[subnet]] is not false.
+     * Example: `10.0.1.5` will normalized to `10.0.1.5/32`, `2008:db0::1` will be normalized to `2008:db0::1/128`
+     * @see subnet
+     */
+    public $normalize = false;
+
+    /**
+     * @var string|boolean
+     * @see negationChar
+     */
+    public $_negationChar = false;
+
+    /**
+     * @var boolean whether to expand an IPv6 address to the full notation format
+     */
+    public $expandV6 = false;
+
+    /**
+     * The property is used by [[isAllowed]] method to determine the order of checks
+     * of the IP address according to [[deny]] and [[allow]] lists.
+     *
+     * When [[order]] is [[ORDER_ALLOW_DENY]] - first all the [[allow]] will be checked, and at least one must
+     * match or the validation will fail. Then all the [[deny]] are checked, if one of them
+     * matched the validation will fail.
+     * At last, if address is not on the [[allow]] nor on the [[deny]] the validation will also fail.
+     *
+     * When [[order]] is [[ORDER_DENY_ALLOW]] - method [[isAllowed]] checks all the [[deny]] and the [[allow]].
+     * If the value is on the [[deny]] the validation will fail, unless it is also present on the [[allow]].
+     * If it is not found on any of the lists the validation will be passed.
+     *
+     * Tip: it is useful to use [[ORDER_ALLOW_DENY]], when it is necessary to deny a less specific subnet and
+     * to allow a more specific one. The example below will result in passing `192.168.1.1`,
+     * but `192.168.2.1` will be denied:
+     *
+     * ```
+     * [
+     *      'deny' => ['192.168.0.0/16'],
+     *      'allow' => ['192.168.1.0/24']
+     * ]
+     * ```
+     *
+     * @var int the order of ranges rules. Used by [[isAllowed]]
+     * @see isAllowed()
+     */
+    public $order = self::ORDER_DENY_ALLOW;
```

Comment:
Instead of introducing such options it could be done alike access control filter i.e.

```php
'ips' => [
  'allow' => ['192.168.0.0/16'],
  'deny' => ['192.168.0.0/16'],
]
```
or 

'ips' => [
  'deny' => ['192.168.0.0/16'],
  'allow' => ['192.168.0.0/16'],
]
```